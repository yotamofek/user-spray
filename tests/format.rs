use std::{error::Error, fs};

use insta::{assert_snapshot, glob};

#[test]
fn format() -> Result<(), Box<dyn Error>> {
    use user_spray::format;

    glob!("inputs", "*.stdin", |path| {
        let contents = fs::read_to_string(path).unwrap();
        let mut output = vec![];
        format(&contents, &mut output).unwrap();
        let output = String::from_utf8(output).unwrap();
        assert_snapshot!(output);
    });

    Ok(())
}
