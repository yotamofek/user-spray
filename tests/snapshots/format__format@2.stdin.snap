---
source: tests/format.rs
expression: output
input_file: tests/inputs/2.stdin
snapshot_kind: text
---
mod outer {
    pub mod inner {
        //! inner use items are currently unsupported
        use crate::Hello;
        use crate::World;
        use std::io::Write;
    }
}

struct Hello;
struct World;
